services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -79,52 +79,138 @@ function hasIdColumn(fields, primaryKeys) {
     || _.includes(primaryKeys, 'id');
 }
 
+// NOTICE: Check the foreign key's reference unicity
+function checkUnicity(primaryKeys, uniqueIndexes, columnName) {
+  const isUnique = uniqueIndexes !== null
+    && uniqueIndexes.find((indexColumnName) =>
+      indexColumnName.length === 1 && indexColumnName.includes(columnName));
+
+  const isPrimary = _.isEqual([columnName], primaryKeys);
+  return { isPrimary, isUnique };
+}
+
+// NOTICE: Format the references depending on the type of the association
+function createReference(foreignKey, association, manyToManyForeignKey) {
+  const reference = {
+    foreignKey: foreignKey.columnName,
+    foreignKeyName: _.camelCase(foreignKey.columnName),
+    association,
+  };
+
+  if (association === 'belongsTo') {
+    reference.ref = foreignKey.foreignTableName;
+    reference.as = formatAliasName(foreignKey.columnName);
+  } else if (association === 'belongsToMany') {
+    reference.ref = manyToManyForeignKey.foreignTableName;
+    reference.otherKey = manyToManyForeignKey.columnName;
+    reference.junctionTable = foreignKey.tableName;
+  } else {
+    reference.ref = foreignKey.tableName;
```

Comment:
We also need to declare aliases here.
Without it you can have the following errors:
<img width="287" alt="Screenshot 2019-12-17 at 08 33 19" src="https://user-images.githubusercontent.com/664261/70974576-5d4ad000-20a8-11ea-8a48-252f559e202f.png">
![Uploading Screenshot 2019-12-17 at 08.33.28.png…]()
![Uploading Screenshot 2019-12-17 at 08.33.31.png…]()